### PR TITLE
build tests for metadata and rec article funcs

### PR DESCRIPTION
**Pull request by fkakatie**

### CODE REVIEW

**Comment by fkakatie** [comment]
I am aware I will need to refactor these once I make use of the `createTable` functionality...

File: package.json [hunk 1]
```diff
@@ -11,8 +11,9 @@
   },
   "devDependencies": {
     "@types/node": "14.14.10",
+    "eslint": "^7.32.0",
     "mock-fs": "4.13.0",
-    "tslint": "6.1.3"
+    "tslint": "^6.1.3"
```
Comment (kptdobe) [inline package.json:16]: We should use strict dependencies (and potential install Renovate).

File: src/blogtoblog/BlogToBlogImporter.ts [hunk 2]
```diff
@@ -138,27 +137,34 @@ export default class BlogToBlogImporter extends PageImporter {
...
       category = topicsArr.shift();
-      topics = topicsArr.join(', ');
       const categoryRow = document.createElement('tr');
       table.append(categoryRow);
       const categoryTitle = document.createElement('td');
       categoryTitle.textContent = 'Category';
       categoryRow.append(categoryTitle);
       const categoryData = document.createElement('td');
-      categoryData.textContent = category;
+      categoryData.textContent = category.trim();
```
Comment (kptdobe) [inline src/blogtoblog/BlogToBlogImporter.ts:146]: Nitpicking: Why adding the `trim()` here and not when setting the `category` value ? There is already a `shift()` there

File: src/blogtoblog/BlogToBlogImporter.ts [hunk 3]
```diff
@@ -102,7 +102,7 @@ export default class BlogToBlogImporter extends PageImporter {
       authorTitle.textContent = 'Author';
       authorRow.append(authorTitle);
       const authorData = document.createElement('td');
-      authorData.textContent = author;
+      authorData.textContent = author.trim();
```
Comment (kptdobe) [inline src/blogtoblog/BlogToBlogImporter.ts:105]: Nitpicking: why adding the `trim()` here and not when setting the `author` value ? There is already a `replace()` there...

File: src/blogtoblog/BlogToBlogImporter.ts [hunk 4]
```diff
@@ -113,7 +113,7 @@ export default class BlogToBlogImporter extends PageImporter {
       dateTitle.textContent = 'Publication Date';
       dateRow.append(dateTitle);
       const dateData = document.createElement('td');
-      dateData.textContent = date;
+      dateData.textContent = date.trim();
```
Comment (kptdobe) [inline src/blogtoblog/BlogToBlogImporter.ts:116]: Nitpicking: why adding the `trim()` here and not when setting the `date` value ? There is already a `replace()` there...

File: src/blogtoblog/BlogToBlogImporter.ts [hunk 5]
```diff
@@ -138,27 +137,34 @@ export default class BlogToBlogImporter extends PageImporter {
...
+      topics = topicsArr.join(', ');
+      if (topicsArr[0] !== 'undefined') {
         const topicsRow = document.createElement('tr');
         table.append(topicsRow);
         const topicsTitle = document.createElement('td');
         topicsTitle.textContent = 'Topics';
         topicsRow.append(topicsTitle);
         const topicsData = document.createElement('td');
-        topicsData.textContent = topics;
+        topicsData.textContent = topics.trim();
```
Comment (kptdobe) [inline src/blogtoblog/BlogToBlogImporter.ts:156]: Same.

File: test/blogtoblog/BlogToBlogImport.spec.ts [hunk 6]
```diff
@@ -39,4 +40,73 @@ describe('BlogToBlogImporter#convertBlocksToTables tests', () => {
...
+      `<main>${div}${div}${div}${div}<table><tr><th>recommended articles</th></tr><tr><td><a href="https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air">https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air</a>
+<a href="https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air">https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air</a>
+<a href="https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air">https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air</a>
+</td></tr></table>${div}</main>`);
+  });
+  it('buildRecommendedArticlesTable unformatted title input', () => {
+    test(
+      `<main><div><h2>featured posts  </h2><ul><li><a href="https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air">https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air</a></li></ul></div></main>`,
+      `<main><table><tr><th>recommended articles</th></tr><tr><td><a href="https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air">https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air</a>
```
Comment (kptdobe) [inline test/blogtoblog/BlogToBlogImport.spec.ts:64]:
```suggestion
      `<main><table><tr><th>Recommended Articles</th></tr><tr><td><a href="https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air">https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air</a>
```

**Review by kptdobe — CHANGES_REQUESTED** [review]

File: test/blogtoblog/BlogToBlogImport.spec.ts [hunk 7]
```diff
@@ -12,14 +12,106 @@
 
 import BlogToBlogImporter from '../../src/blogtoblog/BlogToBlogImporter';
 
+import Blocks from '../../src/utils/Blocks';
+
 import { strictEqual } from 'assert';
 import { describe, it } from "mocha";
 
 import { JSDOM } from 'jsdom';
+import { createStrictEquality } from 'typescript';
```
Comment (kptdobe) [inline test/blogtoblog/BlogToBlogImport.spec.ts:21]: Used, please remove.

File: test/blogtoblog/BlogToBlogImport.spec.ts [hunk 8]
```diff
@@ -12,14 +12,106 @@
...
+    getImporter().buildMetadataTable(document, document);
+    strictEqual(document.body.innerHTML.trim(), expected.trim());
+  };
+
+  const div = `<div></div>`;
+
+  it('build metadata table with expected input', () => {
+    test(
+      `<main>${div}${div}<div><p>By Katie Sexton</p><p>Posted on 09-09-2019</p></div>${div}<div><p>Topics: Alpha, Beta, Gamma,</p><p>Products: Delta, Echo, Foxtrot,</p></div></main>`,
+      `<main>${div}${div}${div}<table><tr><th>metadata</th></tr><tr><td>Author</td><td>Katie Sexton</td></tr><tr><td>Publication Date</td><td>09-09-2019</td></tr><tr><td>Category</td><td>Alpha</td></tr><tr><td>Topics</td><td>Beta, Gamma, Delta, Echo, Foxtrot</td></tr></table></main>`);
```
Comment (kptdobe) [inline test/blogtoblog/BlogToBlogImport.spec.ts:90]:
```suggestion
      `<main>${div}${div}${div}<table><tr><th>Metadata</th></tr><tr><td>Author</td><td>Katie Sexton</td></tr><tr><td>Publication Date</td><td>09-09-2019</td></tr><tr><td>Category</td><td>Alpha</td></tr><tr><td>Topics</td><td>Beta, Gamma, Delta, Echo, Foxtrot</td></tr></table></main>`);
```

**Review by kptdobe — CHANGES_REQUESTED** [review]

File: src/blogtoblog/BlogToBlogImporter.ts [hunk 9]
```diff
@@ -125,18 +123,26 @@ export default class BlogToBlogImporter extends PageImporter {
       const categoryData = document.createElement('td');
       categoryData.textContent = category;
       categoryRow.append(categoryData);
-      if (topics.length) {
+      topics = topicsArr.join(', ');
+      if (topicsArr[0] !== 'undefined') {
```
Comment (kptdobe) [inline src/blogtoblog/BlogToBlogImporter.ts:127]: How can this be ? Having a string with `undefined` value is a bad practice, we should catch this earlier.

File: src/blogtoblog/BlogToBlogImporter.ts [hunk 10]
```diff
@@ -125,18 +123,26 @@ export default class BlogToBlogImporter extends PageImporter {
...
         const topicsRow = document.createElement('tr');
         table.append(topicsRow);
         const topicsTitle = document.createElement('td');
         topicsTitle.textContent = 'Topics';
         topicsRow.append(topicsTitle);
         const topicsData = document.createElement('td');
-        topicsData.textContent = topics;
+        topicsData.textContent = topics
```
Comment (kptdobe) [inline src/blogtoblog/BlogToBlogImporter.ts:134]: why ? ;)

**Review by kptdobe — CHANGES_REQUESTED** [review]

File: src/blogtoblog/BlogToBlogImporter.ts (hunk 10 above)
Comment (fkakatie) [reply src/blogtoblog/BlogToBlogImporter.ts:134]: Because I'm struggling deeply.

File: src/blogtoblog/BlogToBlogImporter.ts (hunk 9 above)
Comment (fkakatie) [reply src/blogtoblog/BlogToBlogImporter.ts]: If the products and topics only have one item between then, that item will become the category, no longer a product or topic. But agreed and fixed.

**Review by kptdobe — APPROVED** [review]

**Pull request merged**